providers/unboundid/src/java/org/sakaiproject/unboundid/UnboundidDirectoryProvider.java

```diff
@@ -303,9 +303,10 @@ protected synchronized boolean createConnectionPool() {
 
                 // Set some sane defaults to better handle timeouts. Unboundid will wait 30 seconds by default on a hung connection.
                 LDAPConnectionOptions connectOptions = new LDAPConnectionOptions();
-                connectOptions.setAbandonOnTimeout(true);
+                connectOptions.setAbandonOnTimeout(false); // If no response from server, dont send an abandon request to the server
                 connectOptions.setConnectTimeoutMillis(operationTimeout);
                 connectOptions.setResponseTimeoutMillis(operationTimeout); // Sakai should not be making any giant queries to LDAP
+                connectOptions.setUseSynchronousMode(true); // "operate more efficiently and without requiring a separate reader thread per connection"
```

Comment:
@bjones86 this might matter to you .... should be faster mode

Reply:
Thanks, @ottenhoff. We're planning to continue to use JLDAP in 20, is this only relevant for unbound?

Reply:
not relevant to JLDAP